v2/cmd/asoctl/internal/importing/importable_arm_resource.go

```diff
@@ -94,11 +95,44 @@ func (i *importableARMResource) Resource() genruntime.MetaObject {
 // ctx is the context to use for the import.
 // Returns a slice of child resources needing to be imported (if any), and/or an error.
 // Both are returned to allow returning partial results in the case of a partial failure.
-func (i *importableARMResource) Import(ctx context.Context, bar *mpb.Bar) ([]ImportableResource, error) {
-	var ref genruntime.ResourceReference
-	ref, err := i.importResource(ctx, i.armID)
+func (i *importableARMResource) Import(ctx context.Context, bar *mpb.Bar) error {
+	// Create an importable blank object into which we capture the current state of the resource
+	importable, err := i.createImportableObjectFromID(i.owner, i.armID)
+	if err != nil {
+		// Error doesn't need additional context
+		return err
+	}
+
+	loader := i.createImportFunction(importable)
+	result, err := loader(ctx, importable, i.owner)
 	if err != nil {
-		return nil, err
+		return err
+	}
+
+	if because, skipped := result.Skipped(); skipped {
+		gk := importable.GetObjectKind().GroupVersionKind().GroupKind()
+		return NewImportSkippedError(gk, i.armID.Name, because)
+	}
+
+	i.resource = importable
+	bar.SetCurrent(1)
+
+	return nil
+}
+
+func (i *importableARMResource) FindChildren(ctx context.Context, bar *mpb.Bar) ([]ImportableResource, error) {
+	if i.resource == nil {
+		// Nothing to do
+		return nil, nil
```

Comment:
Is this `nil` as `[]ImportableResource` handled somewhere?

Reply:
There's no difference - both are empty slices.